static/js/list_render.js

```diff
@@ -93,7 +93,7 @@ exports.create = function ($container, list, opts) {
             const slice = meta.filtered_list.slice(meta.offset, meta.offset + load_count);
 
             const finish = blueslip.start_timing('list_render ' + opts.name);
-            const html = _.reduce(slice, function (acc, item) {
+            const html = slice.map(item => {
```

Comment:
I wanted to make sure that reduce() wasn't actually faster here, so I ran some numbers:

~~~
timeit = (f) => { t = performance.now(); f(); f(); f(); f(); f(); t2 = performance.now(); console.info(t2 - t); }
lst = people.get_active_human_persons().map((p) => p.full_name)
f1 = () => lst.map((i) => i.toString()).join('')
f2 = () => lst.reduce((acc, i) => acc + i.toString())
~~~

The newer map idiom is about 2x as fast when I tried it in FF.

Reply:
If nothing else, I think we can tone down the commit message.  Instead of saying "good for making you feel clever", which seems like a vague insult, can we just say the "idiom is hard to understand and overrated for performance"?  (In fairness I've written similar commit messages about idioms I hate, and like I said, I do generally hate reduce, so I'm glad it's gone.)

Reply:
This function isn’t fast because it’s micro-optimized (it’s…really not), but because it’s called with small enough values of `load_count` (2, 20, or 80) that such micro-optimization can’t possibly matter.

Nonetheless, as long as we are microbenchmarking it, I’ve switched it to an idiom that microbenchmarks just as well.